sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeActionsAsyncClient.java

```diff
@@ -1199,42 +1201,45 @@ private AnalyzeActionsResult toAnalyzeActionsResultLanguageApi(AnalyzeTextJobSta
         }
 
         //TODO: In Language REST API, there is no such task names. It might be a different way to parse the Error Target
+        // https://github.com/Azure/azure-sdk-for-java/issues/28834
         final List<Error> errors = analyzeJobState.getErrors();
         if (!CoreUtils.isNullOrEmpty(errors)) {
             for (Error error : errors) {
-                final String[] targetPair = parseActionErrorTarget(error.getTarget());
-                final String taskName = targetPair[0];
-                final Integer taskIndex = Integer.valueOf(targetPair[1]);
-                final TextAnalyticsActionResult actionResult;
-                if (ENTITY_RECOGNITION_TASKS.equals(taskName)) {
-                    actionResult = recognizeEntitiesActionResults.get(taskIndex);
-                } else if (ENTITY_RECOGNITION_PII_TASKS.equals(taskName)) {
-                    actionResult = recognizePiiEntitiesActionResults.get(taskIndex);
-                } else if (KEY_PHRASE_EXTRACTION_TASKS.equals(taskName)) {
-                    actionResult = extractKeyPhrasesActionResults.get(taskIndex);
-                } else if (ENTITY_LINKING_TASKS.equals(taskName)) {
-                    actionResult = recognizeLinkedEntitiesActionResults.get(taskIndex);
-                } else if (SENTIMENT_ANALYSIS_TASKS.equals(taskName)) {
-                    actionResult = analyzeSentimentActionResults.get(taskIndex);
-                } else if (EXTRACTIVE_SUMMARIZATION_TASKS.equals(taskName)) {
-                    actionResult = extractSummaryActionResults.get(taskIndex);
-                } else if (CUSTOM_ENTITY_RECOGNITION_TASKS.equals(taskName)) {
-                    actionResult = recognizeCustomEntitiesActionResults.get(taskIndex);
-                } else if (CUSTOM_SINGLE_CLASSIFICATION_TASKS.equals(taskName)) {
-                    actionResult = singleCategoryClassifyActionResults.get(taskIndex);
-                } else if (CUSTOM_MULTI_CLASSIFICATION_TASKS.equals(taskName)) {
-                    actionResult = multiCategoryClassifyActionResults.get(taskIndex);
-                } else {
-                    throw logger.logExceptionAsError(new RuntimeException(
-                        "Invalid task name in target reference, " + taskName));
-                }
+                if (error != null) {
+                    final String[] targetPair = parseActionErrorTarget(error.getTarget(), error.getMessage());
+                    final String taskName = targetPair[0];
+                    final Integer taskIndex = Integer.valueOf(targetPair[1]);
```

Comment:
Should we check if this is Null to avoid NPE?

Reply:
Will add below code in the parseActionErrorTarget 
```java
        if (CoreUtils.isNullOrEmpty(taskNameIdPair[0])) {
            throw logger.logExceptionAsError(new RuntimeException("Missing task name in the target reference"));
        }

        if (CoreUtils.isNullOrEmpty(taskNameIdPair[1])) {
            throw logger.logExceptionAsError(new RuntimeException("Missing task id in the target reference"));
        }
```